src/v/kafka/server/connection_context.cc

```diff
@@ -821,8 +821,7 @@ connection_context::client_protocol_state::do_process_responses(
     auto msg = response_as_scattered(std::move(resp_and_res.response));
     if (resp_and_res.resources->request_data.request_key == fetch_api::key) {
         co_await connection_ctx->_server.quota_mgr().record_fetch_tp(
-          resp_and_res.resources->request_data.client_id.value_or(""),
-          msg.size());
+          resp_and_res.resources->request_data.client_id, msg.size());
```

Comment:
The previous commit tried to be just a refactor, so because it changed the type of `request_data.client_id` from `ss::sstring` to `std::optional<ss::sstring>`, to maintain the behaviour I had to do the `value_or` here. The next commit then removes this `value_or`. I'll fold this in though, I mainly just wanted to make this change apparent.